crates/red_knot_python_semantic/src/types/unpacker.rs

```diff
@@ -35,16 +35,25 @@ impl<'db> Unpacker<'db> {
         self.context.db()
     }
 
-    /// Unpack the value to the target expression.
     pub(crate) fn unpack(&mut self, target: &ast::Expr, value: UnpackValue<'db>) {
         debug_assert!(
             matches!(target, ast::Expr::List(_) | ast::Expr::Tuple(_)),
-            "Unpacking target must be a list or tuple expression"
+            "Unpacking target must be a list or tuple"
         );
-
         let mut value_ty = infer_expression_types(self.db(), value.expression())
             .expression_ty(value.scoped_expression_id(self.db(), self.scope));
 
+        let is_in_stub_file = value
+            .expression()
+            .file(self.db())
+            .is_stub(self.db().upcast());
+        let value_is_ellipsis_literal = value
+            .expression()
+            .node_ref(self.db())
+            .is_ellipsis_literal_expr();
+        if value.is_assign() && is_in_stub_file && value_is_ellipsis_literal {
+            value_ty = Type::Unknown;
+        }
```

Comment:
You're doing a bit more work here than you need to if it's _not_ a stub file, because all these conditions are evaluated whether or not it's a stub file. You can take advantage of boolean short-circuiting if you just use one combined test rather than assigning the intermediate variables:

```suggestion
        if value.is_assign()
            && self.context.in_stub()
            && value
                .expression()
                .node_ref(self.db())
                .is_ellipsis_literal_expr()
        {
            value_ty = Type::Unknown;
        }

```

Reply:
Oh the other change made this outdated. I apply this manually.